aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -342,6 +352,75 @@ module aptos_framework::account {
+        assert!(account_scheme == 0 || account_scheme == 1, error::invalid_argument(EINVALID_SCHEME));
+
+        let account_resource = borrow_global_mut<Account>(addr);
+        let signer_capability_offer_proof_challenge = SignerCapabilityOfferProofChallenge {
+            sequence_number: account_resource.sequence_number,
+            recipient_address,
+        };
+
+        if (account_scheme == ED25519_SCHEME) {
+            let pubkey = ed25519::new_unvalidated_public_key_from_bytes(account_public_key_bytes);
+            let matching_auth_key_to_public_key_bytes = ed25519::unvalidated_public_key_to_authentication_key(&pubkey);
+            assert!(account_resource.authentication_key == matching_auth_key_to_public_key_bytes, error::invalid_argument(EWRONG_CURRENT_PUBLIC_KEY));
+
+            let signer_capability_sig = ed25519::new_signature_from_bytes(signer_capability_sig_bytes);
+            assert!(ed25519::signature_verify_strict_t(&signer_capability_sig, &pubkey, signer_capability_offer_proof_challenge), error::invalid_argument(EINVALID_PROOF_OF_KNOWLEDGE));
+            if (option::is_some(&account_resource.signer_capability_offer.for)) {
+                option::swap(&mut account_resource.signer_capability_offer.for, recipient_address);
+            } else {
+                option::fill(&mut account_resource.signer_capability_offer.for, recipient_address);
+            }
+        }
+        else {
+            let pubkey = multi_ed25519::new_unvalidated_public_key_from_bytes(account_public_key_bytes);
+            let matching_auth_key_to_public_key_bytes = multi_ed25519::unvalidated_public_key_to_authentication_key(&pubkey);
+            assert!(account_resource.authentication_key == matching_auth_key_to_public_key_bytes, error::invalid_argument(EWRONG_CURRENT_PUBLIC_KEY));
+
+            let signer_capability_sig = multi_ed25519::new_signature_from_bytes(signer_capability_sig_bytes);
+            assert!(multi_ed25519::signature_verify_strict_t(&signer_capability_sig, &pubkey, signer_capability_offer_proof_challenge), error::invalid_argument(EINVALID_PROOF_OF_KNOWLEDGE));
+            if (option::is_some(&account_resource.signer_capability_offer.for)) {
+                option::swap(&mut account_resource.signer_capability_offer.for, recipient_address);
+            } else {
+                option::fill(&mut account_resource.signer_capability_offer.for, recipient_address);
+            };
+        }
+    }
+
+    public fun accept_signer_capability(account: &signer, offerer_address: address) acquires Account, SignerAuthority {
+        assert!(exists_at(offerer_address), error::not_found(EACCOUNT_DOES_NOT_EXIST));
+
+        let account_resource = borrow_global_mut<Account>(offerer_address);
+        let addr = signer::address_of(account);
+        assert!(option::contains(&account_resource.signer_capability_offer.for, &addr), error::invalid_argument(EINVALID_ACCEPT_CAPABILITY_OFFER));
```

Comment:
we don't need this anymore. acquire_signer should only need to use SignerAuthority